src/common/snippets/src/lowered/pass/init_live_ranges.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/lowered/pass/init_live_ranges.hpp"
+#include "snippets/itt.hpp"
+#include "snippets/op/subgraph.hpp"
+#include "snippets/lowered/expressions/buffer_expression.hpp"
+
+#include "snippets/lowered/pass/serialize_control_flow.hpp"
+#include "snippets/lowered/pass/serialize_data_flow.hpp"
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+namespace {
+// Expressions that don't affect lifetime of registers, e.g. Buffer or RankNormalization
+inline bool pass_through_expr(const ExpressionPtr& expr) {
+    const auto& node = expr->get_node();
+    return op::Subgraph::is_shape_infer_op(node)
+#ifdef SNIPPETS_DEBUG_CAPS
+            || ov::is_type<op::PerfCountBeginBase>(node)
+            || ov::is_type<op::PerfCountEndBase>(node)
+#endif
+            || ov::is_type<BufferExpression>(expr);
+}
+
+} // namespace
+
+bool InitLiveRanges::run(LinearIR& linear_ir) {
+    OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::InitLiveRanges")
+    std::map<RegType, size_t> reg_counter;
+
+    // Note: map expiring time to register
+    std::map<double, std::set<Reg>> regs_to_expire;
+
+    for (auto expr_it = linear_ir.begin(); expr_it != linear_ir.end(); expr_it++) {
+        const auto& expr = *expr_it;
+        const auto op = expr->get_node();
+        if (pass_through_expr(expr)) {
+            expr->set_live_regs(std::prev(expr_it)->get()->get_live_regs());
+            continue;
+        }
+
+        OPENVINO_ASSERT(expr->get_output_count() == op->get_output_size() ||
+                        ov::is_type<op::LoopEnd>(op) ||
+                        ov::is_type<ov::op::v0::Result>(op), "Incorrect count of output port descriptors!");
```

Comment:
Could we move it to `Validate` pass? I think it should common validation check in validation-related pass